File: src/main/java/controller/OrderController.java
package controller;

import dao.OrderDao;

import model.Order;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;

@RestController
public class OrderController {

    private OrderDao dao;

    public OrderController(OrderDao repository) {
        this.dao = repository;
    }

    @PostMapping("orders")
    public Order createOrder(@RequestBody @Valid Order order) {
        return dao.save(order);
    }

    @GetMapping(value = "orders", produces = MediaType.APPLICATION_JSON_VALUE)
    public List<Order> getAllOrders() {
        return dao.getAllOrders();
    }

    @GetMapping("orders/{id}")
    public Order getOrder(@PathVariable Long id) {
        return dao.findOrderById(id);
    }

    @DeleteMapping("orders/{id}")
    public void deleteOrder(@PathVariable Long id) {
        dao.deleteOrderById(id);
    }

}

File: src/main/kotlin/controller/OrderController.kt
package controller

import dao.OrderDao
import order.Order
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
class OrderController(private val dao: OrderDao) {
    @PostMapping("orders")
    fun createOrder(@RequestBody @Valid order: Order?): Order {
        return dao.insertOrder(order!!)
    }

    @GetMapping("orders")
    fun getOrders(): List<Order> {
       return dao.allOrders;
    }

    @GetMapping("orders/{id}")
    fun getOrder(@PathVariable id: Long?): Order {
        return dao.findOrderById(id!!)!!
    }

    @DeleteMapping("orders/{id}")
    fun deleteOrder(@PathVariable id: Long?) {
        dao.deleteOrderById(id!!)
    }
}